src/script/script.h

```diff
@@ -118,7 +118,9 @@ enum opcodetype
     OP_EQUAL = 0x87,
     OP_EQUALVERIFY = 0x88,
     OP_RESERVED1 = 0x89,
+    OP_NOTEQUAL = OP_RESERVED1,
     OP_RESERVED2 = 0x8a,
+    OP_NOTEQUALVERIFY = OP_RESERVED2,
```

Comment:
Is this worth taking up an entire opcode slot?

Reply:
I don't know, honestly. You take that logic to the extreme and basically all of Forth can be reduced to a small handful of stack manipulation combinators. Maybe we should have 4-bit opcodes compressed two to a byte, and surrogate pairs for complex validation functions?

I have noted elsewhere that `NOTEQUALVERIFY` would save 2 bytes from the hash collision bounty scripts, if it were available. So at least there's a demonstrated use case. And then of course there's the original intent arguments :P (`RESERVED1` and `RESERVED2` were these opcodes originally, AFAICT.)

We should assess how many opcode bytes we have left to work with and how many we want to keep for special purposes.

Reply:
Perhaps "VERIFY" should be a single bit and be useable by everything (except e.g. OP_IF and such). I don't know if that would save us any space though.

Maybe a "NOT" bit could exist too. So there would be OP_EQUAL, OP_IF, and you could attach  VERIFY/NOT flags to produce OP_EQUALVERIFY, OP_NOTEQUALVERIFY, OP_IFNOT, etc.

Reply:
It'd be a bit wasteful though. I think things which have all those variants are the exception not the norm, and mostly limited to the basic opcodes.

Reply:
Yeah, scratch that idea.